lottie/src/main/java/com/airbnb/lottie/utils/OffscreenLayer.java

```diff
@@ -0,0 +1,494 @@
+package com.airbnb.lottie.utils;
+
+import android.graphics.Bitmap;
+import android.graphics.BlendMode;
+import android.graphics.BlurMaskFilter;
+import android.graphics.Canvas;
+import android.graphics.ColorFilter;
+import android.graphics.Matrix;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffColorFilter;
+import android.graphics.PorterDuffXfermode;
+import android.graphics.Rect;
+import android.graphics.RectF;
+import android.graphics.RenderEffect;
+import android.graphics.RenderNode;
+import android.graphics.Shader;
+import android.os.Build;
+import androidx.annotation.Nullable;
+import androidx.core.graphics.BlendModeCompat;
+import androidx.core.graphics.PaintCompat;
+import com.airbnb.lottie.animation.LPaint;
+
+/**
+ * An OffscreenLayer encapsulates a "child surface" onto which canvas draw calls can be issued.
+ * At the end, the result of these draw calls will be composited onto the parent surface, with
+ * user-provided alpha, blend mode, color filter, and drop shadow.
+ *
+ * To use the OffscreenLayer, call its start() method with the necessary parameters, draw
+ * to the returned canvas, and then call finish() to composite the result onto the main canvas.
+ *
+ * In this sense, using an OffscreenLayer is very similar to the Canvas.saveLayer() family
+ * of functions, and in fact forwards to Canvas.saveLayer() when appropriate.
+ *
+ * Unlike Canvas.saveLayer(), an OffscreenLayer also supports compositing with a drop-shadow,
+ * and uses a hardware-accelerated target when available. It attempts to choose the fastest
+ * approach to render and composite the contents, up to and including simply rendering
+ * directly, if possible.
+ */
+public class OffscreenLayer {
```

Comment:
When I checked out this PR, Android Studio warns of a bunch of nullability issues. They're probably fine in practice but I aim to have all files have no warnings in Android Studio so that new issues are easy to spot. Could you try to tweak the code so that there aren't the nullability issues?

Often, it happens when you do something like:
```
@Nullable Bitmap bitmp…

void foo() {
    if (bitmap != null) {
        bitmap.something()
    }
```

Another thread could technically set bitmap to null so I would do:

```
void foo() {
    Bitmap bitmap = this.bitmap;
    if (bitmap != null) …
}
```

That way your local copy is guaranteed to not change.

Reply:
I tried to fix all issues and I think now all of the warnings are pre-existing. Let me know if I missed something!